mqtt/mqtt-bridge/src/client.rs

```diff
@@ -436,18 +480,22 @@ pub struct UpdateSubscriptionHandle(mqtt3::UpdateSubscriptionHandle);
 
 #[automock]
 impl UpdateSubscriptionHandle {
-    pub async fn subscribe(
-        &mut self,
-        subscribe_to: SubscribeTo,
-    ) -> Result<(), UpdateSubscriptionError> {
-        self.0.subscribe(subscribe_to).await
+    pub async fn subscribe(&mut self, subscribe_to: SubscribeTo) -> Result<(), ClientError> {
+        validate_length(&subscribe_to.topic_filter)?;
```

Comment:
just curious why it is important to validate for such, it seems, edge case? what would happen with the client if we send a long string to it? I assume it will fail in the codec. Is it unrecoverable error?

Reply:
This is a good point actually. We were focused on the client throwing an error and should we re-create client/pump/bridge. And it seems like it is fine to just log an error in this case. I wonder if we even need this input validation. Lets discuss offline?

Reply:
If we don't validate the client will get into an error and can't recover without recreating it. I think we can have both validation and re-create of bridge.

Reply:
actually not needed for subscribe and publish because handle validate internally before sending to the client